File: app/src/main/java/com/itmoldova/photoview/PhotoViewActivity.java
package com.itmoldova.photoview;

import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.widget.TextView;

import com.itmoldova.Extra;
import com.itmoldova.R;
import com.itmoldova.adapter.PhotoViewAdapter;
import com.itmoldova.anim.DepthPageTransformer;

import java.util.ArrayList;

public class PhotoViewActivity extends AppCompatActivity implements ViewPager.OnPageChangeListener {

    private ViewPager viewPager;
    private TextView pageNumber;

    private int totalNumberOfUrls;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_photo_view);

        viewPager = (ViewPager) findViewById(R.id.viewpager);
        pageNumber = (TextView) findViewById(R.id.page_number);

        ArrayList<String> urls = getIntent().getStringArrayListExtra(Extra.INSTANCE.getPHOTO_URLS());
        String clickedUrl = getIntent().getStringExtra(Extra.INSTANCE.getCLICKED_URL());
        totalNumberOfUrls = urls.size();

        PhotoViewAdapter adapter = new PhotoViewAdapter(this, urls);
        viewPager.setAdapter(adapter);
        int currentItem = urls.indexOf(clickedUrl) != -1 ? urls.indexOf(clickedUrl) : 0;
        viewPager.setCurrentItem(currentItem);
        viewPager.addOnPageChangeListener(this);
        viewPager.setPageTransformer(true, new DepthPageTransformer());
        updatePageCounter(currentItem);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowTitleEnabled(false);
    }

    private void updatePageCounter(int currentItem) {
        pageNumber.setText(getString(R.string.page_counter, currentItem + 1, totalNumberOfUrls));
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        viewPager.removeOnPageChangeListener(this);
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
        // not used
    }

    @Override
    public void onPageSelected(int position) {
        updatePageCounter(position);
    }

    @Override
    public void onPageScrollStateChanged(int state) {
        // not used
    }
}

File: app/src/main/java/com/itmoldova/photoview/PhotoViewActivity.kt
package com.itmoldova.photoview

import android.os.Bundle
import android.view.MenuItem
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.viewpager.widget.ViewPager
import com.itmoldova.Extra
import com.itmoldova.R
import com.itmoldova.adapter.PhotoViewAdapter
import com.itmoldova.anim.DepthPageTransformer

class PhotoViewActivity : AppCompatActivity(), ViewPager.OnPageChangeListener {

    private lateinit var viewPager: ViewPager
    private lateinit var pageNumber: TextView

    private var totalNumberOfUrls: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_photo_view)

        viewPager = findViewById(R.id.viewpager)
        pageNumber = findViewById(R.id.page_number)

        val urls = intent.getStringArrayListExtra(Extra.PHOTO_URLS) ?: return
        val clickedUrl = intent.getStringExtra(Extra.CLICKED_URL) ?: return
        totalNumberOfUrls = urls.size

        val currentItem = if (urls.indexOf(clickedUrl) != -1) urls.indexOf(clickedUrl) else 0

        viewPager.adapter = PhotoViewAdapter(this, urls)
        viewPager.currentItem = currentItem
        viewPager.addOnPageChangeListener(this)
        viewPager.setPageTransformer(true, DepthPageTransformer())
        updatePageCounter(currentItem)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowTitleEnabled(false)
    }

    private fun updatePageCounter(currentItem: Int) {
        pageNumber.text = getString(R.string.page_counter, currentItem + 1, totalNumberOfUrls)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onDestroy() {
        super.onDestroy()
        viewPager.removeOnPageChangeListener(this)
    }

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
        // not used
    }

    override fun onPageSelected(position: Int) {
        updatePageCounter(position)
    }

    override fun onPageScrollStateChanged(state: Int) {
        // not used
    }
}
